source/slang/slang-ast-builder.h

```diff
@@ -114,16 +114,21 @@ class ASTBuilder : public RefObject
     {
         union
         {
-            NodeBase* nodeOperand;
-            int64_t intOperand;
+            NodeBase* nodeOperand[2];
+            int64_t intOperand[2];
         } values;
-        NodeOperand() { values.nodeOperand = nullptr; }
-        NodeOperand(NodeBase* node) { values.nodeOperand = node; }
+        NodeOperand()
+        {
+            values.nodeOperand[0] = nullptr;
+            values.nodeOperand[1] = nullptr;
+        }
+        NodeOperand(NodeBase* node) { values.nodeOperand[0] = node; values.nodeOperand[1] = nullptr; }
         template<typename EnumType>
         NodeOperand(EnumType intVal)
         {
             static_assert(sizeof(EnumType) <= sizeof(values), "size of operand must be less than pointer size.");
-            values.intOperand = 0;
+            values.intOperand[0] = 0;
```

Comment:
Hm. Doesn't this assume little endian? If enum type is less than the value int64_t. Wouldn't it be preferable to just use a cast? Care might be needed with sign extension tho.